src/util/JsonUtils.h

```diff
@@ -0,0 +1,51 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <boost/json.hpp>
+
+#include <string>
+
+namespace util {
+
+inline boost::json::object
+removeSecret(boost::json::object const& object)
+{
+    auto newObject = object;
+    auto const secretFields = {"secret", "seed", "seed_hex", "passphrase"};
+
+    if (newObject.contains("params") and newObject.at("params").is_array() and
+        newObject.at("params").as_array().size() > 0 and newObject.at("params").as_array()[0].is_object())
+    {
+        for (auto const& secretField : secretFields)
+        {
+            newObject.at("params").as_array()[0].as_object().erase(secretField);
+        }
+    }
+    // for websocket requests
+    for (auto const& secretField : secretFields)
+    {
+        newObject.erase(secretField);
```

Comment:
I think instead of erasing it would be better to "mask" it. So change the actual secret with something like "<removed>". Just so when something goes wrong we can still see in the log that they really did send the secret field and did not forget to send it.